src/main/java/gyro/google/compute/AutoscalerFinder.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright 2020, Perfect Sense, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package gyro.google.compute;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+import com.google.api.services.compute.Compute;
+import com.google.api.services.compute.model.Autoscaler;
+import com.google.api.services.compute.model.AutoscalerAggregatedList;
+import com.google.api.services.compute.model.AutoscalerList;
+import com.google.api.services.compute.model.AutoscalersScopedList;
+import com.psddev.dari.util.ObjectUtils;
+import com.psddev.dari.util.StringUtils;
+import gyro.core.GyroException;
+import gyro.core.Type;
+import gyro.google.GoogleFinder;
+import gyro.google.util.Utils;
+
+/**
+ * Query an autoscaler.
+ *
+ * You can provide an expression that filters resources. The expression must specify the field name, and the value that you want to use for filtering.
+ *
+ * Please see :doc:`compute-autoscaler` resource for available fields.
+ *
+ * Example
+ * -------
+ *
+ * .. code-block:: gyro
+ *
+ *    autoscaler: $(external-query google::compute-autoscaler { name: 'compute-autoscaler-example', zone: 'us-central1-a' })
+ */
+@Type("compute-autoscaler")
+public class AutoscalerFinder extends GoogleFinder<Compute, Autoscaler, AutoscalerResource> {
+
+    @Override
+    protected List<Autoscaler> findAllGoogle(Compute client) throws Exception {
+        return findAllAutoscalers(client, getProjectId(), ResourceScope.ZONE, null);
+    }
+
+    @Override
+    protected List<Autoscaler> findGoogle(Compute client, Map<String, String> filters) throws Exception {
+        if (filters.containsKey("region")) {
+            throw new GyroException("For regional autoscaler, use 'compute-region-autoscaler' instead.");
+        }
+
+        String zone = filters.remove("zone");
+
+        if (StringUtils.isBlank(zone) || ObjectUtils.isBlank(filters)) {
+            return findAllAutoscalers(client, getProjectId(), ResourceScope.ZONE, filters);
+        }
```

Comment:
The find specific finder should not return everything at any point.
So in this method, `ObjectUtils.isBlank(filters)` is condition will never be true.

Reply:
it doesn't return everything, but still uses the filters a user provided, and what it returns is totally  up to the provider.